### PR TITLE
Armors and suits with storage slots now properly unequip.

### PR DESCRIPTION
**Pull request by DominikPanic**
:cl: LaKiller8
fix: Armors and suits with storage now unequip when you take off your uniform.
/:cl:

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -133,23 +133,24 @@
...
 
 /obj/item/clothing/suit/mob_can_equip(mob/M, slot, disable_warning = 0)
 	//if we can't equip the item anyway, don't bother with other checks.
-	if (!..())
-		return 0
+	. = ..()
+	if(!.)
+		return FALSE
 
 	if(ishuman(M))
```
Comment (spookydonut) [inline code/modules/clothing/clothing.dm]: early return

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -133,23 +133,24 @@
...
 		var/mob/living/carbon/human/H = M
 		var/obj/item/clothing/under/U = H.w_uniform
 		//some uniforms prevent you from wearing any suits but certain types
 		if(U?.suit_restricted && !is_type_in_list(src, U.suit_restricted))
```
Comment (spookydonut) [inline code/modules/clothing/clothing.dm]: early return

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -133,23 +133,24 @@
...
 	//if we can't equip the item anyway, don't bother with other checks.
-	if (!..())
-		return 0
+	. = ..()
+	if(!.)
+		return FALSE
 
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
 		var/obj/item/clothing/under/U = H.w_uniform
```
Comment (spookydonut) [inline code/modules/clothing/clothing.dm]: you need to check this is null

File: code/modules/clothing/clothing.dm [hunk 4]
```diff
@@ -133,23 +133,24 @@
...
 		//some uniforms prevent you from wearing any suits but certain types
 		if(U?.suit_restricted && !is_type_in_list(src, U.suit_restricted))
 			to_chat(H, "<span class='warning'>[src] can't be worn with [U].</span>")
```
Comment (spookydonut) [inline code/modules/clothing/clothing.dm]: runtime error here using [U] without nullcheck

File: code/modules/clothing/clothing.dm (hunk 2 above)
Comment (spookydonut) [reply code/modules/clothing/clothing.dm]: also runtime on U with it not being checked for null

File: code/modules/clothing/clothing.dm [hunk 5]
```diff
@@ -133,23 +133,27 @@
...
-		//some uniforms prevent you from wearing any suits but certain types
-		if(U?.suit_restricted && !is_type_in_list(src, U.suit_restricted))
-			to_chat(H, "<span class='warning'>[src] can't be worn with [U].</span>")
-			return 0
-	return 1
+	. = ..()
+	if(!. || !ishuman(M))
+		return FALSE
+
+	var/mob/living/carbon/human/H = M
```
Comment (Rohesie) [inline code/modules/clothing/clothing.dm]: Change `mob/M` to `mob/living/carbon/human/H` and avoid this typecast.

**Pull request merged**